opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -141,18 +146,41 @@ namespace Opm
                 if (numSamples > 0) {
                     auto splinePvt = std::shared_ptr<PvtDeadSpline>(new PvtDeadSpline);
                     splinePvt->initFromOil(pvdoTables, numSamples);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        splinePvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                     deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = splinePvt;
                 } else {
                     auto deadPvt = std::shared_ptr<PvtDead>(new PvtDead);
                     deadPvt->initFromOil(pvdoTables);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        deadPvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                   deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = deadPvt;
                 }
             } else if (!pvtoTables.empty()) {
-                props_[phase_usage_.phase_pos[Liquid]].reset(new PvtLiveOil(pvtoTables));
```

Comment:
ok, I renamed the `deadPvt` variable to `pvdo` and `splinePvt` to `splinePvdo`. If you want different/better names, just let me know.

for the `shared_ptr` issue: I think that in this case it did not really matter because the raw pointer was immediately converted into a shared one on the next line after construction and the shared pointer took ownership at this point, but you're right that it is better style to use shared pointers from the outset...